### PR TITLE
Add comprehensive tests for create_security_matches filter

### PR DESCRIPTION
**Pull request by cihlamar**
Follow-up to #259
Github Issue: https://github.com/ansible-middleware/amq/issues/248
  Filter improvements:
  - Skip roles with empty permissions
  
  Test suite:
  - Test valid inputs (normal roles, defaults, edge cases)
  - Test invalid inputs (missing name, None, empty string)
  - Test output structure and role appending
  - 12 test cases covering all scenarios

### CODE REVIEW

File: plugins/filter/create_security_matches.py [hunk 1]
```diff
@@ -70,7 +70,7 @@ def create_security_matches(amq_broker_roles):
         match_pattern = role.get('match') or '#'
         permissions = role.get('permissions', [])
 
-        if match_pattern not in result:
+        if match_pattern not in result and permissions:
```
Comment (RanabirChakraborty) [inline plugins/filter/create_security_matches.py]: Better do something like below, It'll only process if there are permissions.

```
  if permissions:                                                                                                 
      if match_pattern not in result:                                                                                                                      
          result[match_pattern] = {} 
```

File: molecule/security_matches_test/converge.yml [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+          register: error_result
+          ignore_errors: true
+
+        - name: Verify error was raised
+          ansible.builtin.assert:
+            that:
+              - error_result is failed
+              - "'requires a list' in error_result.msg | lower"
+            success_msg: "✓ Test 11 passed: Non-list input raises error"
+
```
Comment (RanabirChakraborty) [inline molecule/security_matches_test/converge.yml]: You can also add a test for multiple roles with same match, first has empty permissions.

**Review by RanabirChakraborty — CHANGES_REQUESTED** [review]
Splitting it into `verify.yml` would be much better, there is a lot of changes into one `converge.yml`.  Also add the molecule test name here https://github.com/ansible-middleware/amq/blob/main/.github/workflows/ci.yml#L24-L26 otherwise it'll not run.

File: molecule/security_matches_test/converge.yml [hunk 3]
```diff
@@ -0,0 +1,221 @@
+---
+- name: Test create_security_matches filter
+  hosts: localhost
+  gather_facts: false
+  tasks:
+    - name: "=== TEST 1: Normal role with all fields ==="
```
Comment (hcherukuri) [inline molecule/security_matches_test/converge.yml]: Usually in the converge.yml we just define the scenario and  do the assertions in the verify.yml file. WDYT?

File: plugins/filter/create_security_matches.py (hunk 1 above)
Comment (cihlamar) [reply plugins/filter/create_security_matches.py]: Condition has been modified.

File: molecule/security_matches_test/converge.yml (hunk 2 above)
Comment (cihlamar) [reply molecule/security_matches_test/converge.yml]: The test (12) included.

File: molecule/security_matches_test/converge.yml (hunk 3 above)
Comment (cihlamar) [reply molecule/security_matches_test/converge.yml]: Sounds good to me, the verify.yml file added to assert all expected behaviors.

File: plugins/filter/create_security_matches.py [hunk 4]
```diff
@@ -70,14 +70,15 @@ def create_security_matches(amq_broker_roles):
...
-        if match_pattern not in result:
-            result[match_pattern] = {}
+        if permissions:
+            if match_pattern not in result:
+                result[match_pattern] = {}
 
-        for permission in permissions:
-            if permission not in result[match_pattern]:
-                result[match_pattern][permission] = []
+            for permission in permissions:
```
Comment (RanabirChakraborty) [inline plugins/filter/create_security_matches.py:77]: remove the trailing spaces in the following lines.

**Review by RanabirChakraborty — CHANGES_REQUESTED** [review]
Rebase two commits into one.

File: plugins/filter/create_security_matches.py (hunk 4 above)
Comment (cihlamar) [reply plugins/filter/create_security_matches.py:77]: checked now, no trailing spaces were found.

**Comment by cihlamar** [comment]
> Rebase two commits into one.

Rebased into single commit.

**Review by RanabirChakraborty — CHANGES_REQUESTED** [review]
LGTM!

**Comment by RanabirChakraborty** [comment]
@cihlamar Fixed the CI pipeline, now it's working fine. Please resolve the conflict.

File: molecule/security_matches/molecule.yml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    host_vars:
+      localhost:
+        ansible_connection: local
+        ansible_python_interpreter: "{{ ansible_playbook_python }}"
+  env:
+    ANSIBLE_FORCE_COLOR: "true"
+verifier:
+  name: ansible
+scenario:
+  test_sequence:
```
Comment (RanabirChakraborty) [inline molecule/security_matches/molecule.yml:37]: why `idempotence` is missing?

**Review by RanabirChakraborty — CHANGES_REQUESTED** [review]

File: molecule/security_matches/molecule.yml [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    - cleanup
+    - destroy
+    - create
```
Comment (RanabirChakraborty) [inline molecule/security_matches/molecule.yml:40]: Also you need to add `prepare`, and add prepare.yml like https://github.com/ansible-middleware/amq/blob/main/molecule/live_only/prepare.yml

**Review by RanabirChakraborty — CHANGES_REQUESTED** [review]

File: molecule/security_matches/molecule.yml (hunk 5 above)
Comment (cihlamar) [reply molecule/security_matches/molecule.yml:37]: idempotence added.

File: molecule/security_matches/molecule.yml (hunk 6 above)
Comment (cihlamar) [reply molecule/security_matches/molecule.yml:40]: prepare added.

**Review by RanabirChakraborty — APPROVED** [review]
LGTM
